test/unittests/test_analyse_xml_data/test_extract_links_from_XML.py

```diff
@@ -102,28 +103,54 @@ def test_extract_ids_from_text_prop_with_several_text_links() -> None:
         'href="IRI:res_A_18:IRI">res_A_18</a></text><text permissions="prop-default" encoding="xml"><a '
         'class="salsah-link" href="IRI:res_B_18:IRI">res_B_18</a></text></text-prop>'
     )
-    res = _extract_ids_from_text_prop(test_ele)
-    assert unordered(res) == [{"res_A_18"}, {"res_B_18"}]
+    res = _create_class_instance_text_prop("res_C_18", test_ele)
+    res_ids = [x.object_link_ids for x in res]
+    assert unordered(res_ids) == [{"res_A_18"}, {"res_B_18"}]
 
 
-def test_extract_one_id_resptr_prop() -> None:
+def test_create_class_instance_resptr_link_one_link() -> None:
     test_ele = etree.fromstring(
         '<resptr-prop xmlns="https://dasch.swiss/schema" '
         'xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name=":hasResource1"><resptr '
         'permissions="prop-default">res_C_15</resptr></resptr-prop>'
     )
-    res = _extract_ids_from_one_resptr_prop(test_ele)
-    assert unordered(res) == ["res_C_15"]
+    res = _create_class_instance_resptr_link("res_A_15", test_ele)
+    assert res[0].object_id == "res_C_15"
 
 
-def test_extract_several_id_resptr_prop() -> None:
+def test_create_class_instance_resptr_link_several() -> None:
     test_ele = etree.fromstring(
         '<resptr-prop xmlns="https://dasch.swiss/schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
         'name=":hasResource1"><resptr permissions="prop-default">res_A_13</resptr><resptr '
         'permissions="prop-default">res_B_13</resptr><resptr permissions="prop-default">res_C_13</resptr></resptr-prop>'
     )
-    res = _extract_ids_from_one_resptr_prop(test_ele)
-    assert unordered(res) == ["res_A_13", "res_B_13", "res_C_13"]
+    res = _create_class_instance_resptr_link("res_D_13", test_ele)
+    assert res[0].object_id == "res_A_13"
+    assert res[1].object_id == "res_B_13"
+    assert res[2].object_id == "res_C_13"
```

Comment:
```suggestion
    assert res == ["res_A_13", "res_B_13",  "res_C_13"]
```

Reply:
ah, no, my bad, overlooked that... in that case it's ok. The most thorough thing to do would be to test for the entire class objects, but if you think that's overkill, then leave it as it is now

Reply:
I did it like this because I wanted to test only a certain component.